app/src/main/java/com/example/househomey/SignInFragment.java

```diff
@@ -0,0 +1,200 @@
+     * initialize Firebase Authentication, and set up UI components such as username and password
+     * fields, login button, and redirection text views. Handles user input for login, validates
+     * username and password, and initiates the login process using Firebase Authentication.
+     * Navigates to the Sign Up fragment when the redirection link is clicked.
+     *
+     * @param inflater           The LayoutInflater object that can be used to inflate any views in
+     *                           the fragment.
+     * @param container          If non-null, this is the parent view that the fragment's UI should
+     *                           be attached to. The fragment should not add the view itself, but
+     *                           this can be used to generate the LayoutParams of the view.
+     * @param savedInstanceState If non-null, this fragment is being re-constructed from a previous
+     *                           saved state as given here.
+     * @return The root view of the fragment's layout.
+     */
+    @Override
+    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View rootView = inflater.inflate(R.layout.fragment_user_signin, container, false);
+
+        // Initialize Firebase Auth
+        auth = FirebaseAuth.getInstance();
+        auth.signOut();
+
+        usernameEdittext = rootView.findViewById(R.id.signin_username);
+        passwordEdittext = rootView.findViewById(R.id.signin_password);
+        loginButton = rootView.findViewById(R.id.signin_button);
+        TextView signinRedirectMessage = rootView.findViewById(R.id.signin_redirect_message);
+        TextView signinRedirect = rootView.findViewById(R.id.signin_redirect);
+
+        loginButton.setText(R.string.signin_button_text);
+        signinRedirectMessage.setText(R.string.signin_text);
+        signinRedirect.setText(R.string.sign_up_string);
+
+        loginButton.setOnClickListener(v -> {
+            loginButton.setBackgroundResource(R.drawable.signin_button_clicked);
+            loginButton.setTextColor(getResources().getColor(R.color.creme, rootView.getContext().getTheme()));
+            new Handler(Looper.getMainLooper()).postDelayed(() -> {
+                loginButton.setBackgroundResource(R.drawable.signin_button_unclicked);
+                loginButton.setTextColor(getResources().getColor(R.color.brown, rootView.getContext().getTheme()));
+            }, 150);
+
+            username = usernameEdittext.getText().toString().trim();
+            password = passwordEdittext.getText().toString().trim();
+            usernameEdittext.setError(null);
+            passwordEdittext.setError(null);
+
+
+            userRef = FirebaseFirestore.getInstance().collection("user");
+
+            if (!username.isEmpty() & !password.isEmpty()) {
+                docRef = userRef.document(username);
+                docRef.get().addOnCompleteListener(task -> {
```

Comment:
this can be replaced with `addOnSuccessListener` like we found out for the other one. Should help reduce the IF nesting

Reply:
done